### PR TITLE
5234 Migrate deepvariant pytest to NF test

### LINKED ISSUE #5234: Port deepvariant pytests to nf-test
### Is your feature request related to a problem? Please describe

Part of #3937 

pytest -> nf-test

### Describe the solution you'd like

nf-test tests

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lukbut**
## PR checklist

Closes #5234

Migrated deepvariant Pytest tests to nf-test.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/deepvariant/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,108 @@
+nextflow_process {
+
+    name "Test Process DEEPVARIANT"
+    script "../main.nf"
+    process "DEEPVARIANT"
+
+    tag "modules"
+    tag "modules_nfcore"
+
+    test("Test deepvariant") {
```
Comment (Joon-Klaps) [inline modules/nf-core/deepvariant/tests/main.nf.test:10]: Test names should be a little more descriptive from the [website](https://nf-co.re/docs/contributing/modules#migrating-from-pytest-to-nf-test):
``` 
Provide a test name preferably indicating the test-data and file-format used. Example: test("homo_sapiens - [bam, bai, bed] - fasta - fai")
```

File: modules/nf-core/deepvariant/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                      file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
+                      file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
```
Comment (Joon-Klaps) [inline modules/nf-core/deepvariant/tests/main.nf.test:17]: Rewrite all the input files with `params.modules_testdata_base_path`
```suggestion
                      file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                      file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
```

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
Nice work, minor suggestions.

File: modules/nf-core/deepvariant/tests/main.nf.test (hunk 1 above)
Comment (lukbut) [reply modules/nf-core/deepvariant/tests/main.nf.test]: Modified as suggested

Comment (Joon-Klaps) [reply modules/nf-core/deepvariant/tests/main.nf.test]: it was an example from the docs, it wasn't a suggestion. I'm sorry for being unclear 😖

Comment (lukbut) [reply modules/nf-core/deepvariant/tests/main.nf.test]: Yep realised this and modified it again, no worries!

File: modules/nf-core/deepvariant/tests/main.nf.test (hunk 2 above)
Comment (lukbut) [reply modules/nf-core/deepvariant/tests/main.nf.test:17]: @Joon-Klaps where can I find the files available for testing so I construct the paths, please?

Comment (lukbut) [reply modules/nf-core/deepvariant/tests/main.nf.test:17]: Found it ;) https://github.com/nf-core/test-datasets/tree/modules/data/genomics/homo_sapiens

Comment (Joon-Klaps) [reply modules/nf-core/deepvariant/tests/main.nf.test:17]: All files are in this location https://github.com/nf-core/modules/blob/master/tests/config/test_data.config . 
Note that you'll have to change `params.test_data_base` to `params.modules_testdata_base_path`.

**Review by opetryk — APPROVED** [review]
Looks good for me!

**Review by Joon-Klaps — COMMENTED** [review]
Snapshot still empty

File: modules/nf-core/deepvariant/tests/nextflow-intervals.config [hunk 3]
```diff
@@ -0,0 +1,10 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/nf-core/deepvariant/tests/nextflow-intervals.config:4]: We don't publish, so not needed

**Review by maxulysse — APPROVED** [review]

**Pull request merged**